assets/evaluation_on_cloud/environments/evaluations-built-in/context/evaluate_on_data.py

```diff
@@ -77,12 +101,134 @@ def get_evaluator_config(command_line_args):
     return evaluator_config
 
 
-def run_evaluation(command_line_args, evaluators, evaluator_config):
+def create_model_target_and_data_mapping(command_line_args):
+    """Get model target configuration from user input."""
+    if not command_line_args.eval_target:
+        logger.info("No eval_target provided. Returning None.")
+        return (None, None)
+    try:
+        logger.info("Eval_target provided.")
+        target_config = json.loads(command_line_args.eval_target)
+    except json.JSONDecodeError as e:
+        raise RuntimeError(f"Invalid JSON in eval_target: {e}")
+
+    model_config = target_config.get(MODEL_CONFIG, {})
+
+    api_key_env = model_config.get(API_KEY, "")
+    api_key_value = os.environ.get(api_key_env.upper(), "")
+    if not api_key_value:
+        raise RuntimeError(f"API key environment variable '{api_key_env.upper()}' is missing or empty!")
+
+    model_config_type = str(model_config.get(TYPE, ""))
+    logger.info(f"  - Type: {model_config_type}")
+
+    endpoint = model_config.get(AZURE_ENDPOINT, "")
+    model_params = target_config.get(MODEL_PARAMS, {}).copy()
+    data_mapping = model_params.pop(DATA_MAPPING, None)
+    if data_mapping is None:
+        data_mapping = DEFAULT_DATA_MAPPING
+        logger.info(f"Using default dataMapping: {data_mapping}")
+
+    system_message = target_config.get(SYSTEM_MESSAGE, "")
+    few_shot_examples = target_config.get(FEW_SHOT_EXAMPLES, [])
+
+    logger.info("Creating ModelTarget with values:")
+    logger.info(f"  - Endpoint: {endpoint}")
+    logger.info(f"  - ApiKey: {'[HIDDEN]' if api_key_value else 'MISSING'}")
+    logger.info(f"  - ModelParams: {model_params}")
+    logger.info(f"  - SystemMessage: {system_message}")
+    logger.info(f"  - FewShotExamples: {few_shot_examples}")
+    logger.info(f"  - DataMapping: {data_mapping}")
+
+    model_target = ModelTarget(
+        endpoint=endpoint,
+        api_key=api_key_value,
```

Comment:
Track a workitem for this when we have to add support for EntraID too. 
It could be "Support model target with entraID"